File: e-commerce/urunKatalog4/backend/Entity/Order/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace urunKatalog4.Entity.Order
{
    public class Order
    {
        public int Id { get; set; }
        public string aliciId { get; set; }
        public ShippingAdres ShippingAdres { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.Now;
        public List<OrderItem> OrderItems { get; set; }
        public long Subtotal { get; set; }
        public long DeliveryFee { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Hazırlanıyor;
        public string PaymentIntentId { get; set; }

        public long GetTotal()
        {
            return Subtotal + DeliveryFee;
        }
        
    }
}

File: e-commerce/urunKatalog4/ui/src/app/model/order.ts
export interface ShippingAdres {
    aliciName: string;
    adres1: string;
    adres2: string;
    il: string;
    ilce: string;
    ulke: string;
}

export interface OrderItem {
    urunId: number;
    urunName: string;
    urunImageUrl: string;
    urunFiyat: number;
    quantity: number;
}

export interface Order {
    id: number;
    aliciId: string;
    shippingAdres: ShippingAdres;
    orderDate: string;
    orderItems: OrderItem[];
    subtotal: number;
    deliveryFee: number;
    orderStatus: string;
    paymentIntentId: string;
    total: number;
}
